metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java

```diff
@@ -116,25 +120,39 @@ public void timeParse(int reps) throws Exception {
     }
   }
 
-  private static String resourceToString(String path) throws Exception {
-    InputStream in = ParseBenchmark.class.getResourceAsStream(path);
-    if (in == null) {
-      throw new IllegalArgumentException("No such file: " + path);
+  private static File getResourceFile(String path) throws Exception {
+    URL url = ParseBenchmark.class.getResource("/ParseBenchmarkData.zip");
+    if (url == null) {
+      throw new IllegalArgumentException("Resource " + path + " does not exist");
     }
+    File file = new File(url.toURI());
+    if (!file.isFile()) {
+      throw new IllegalArgumentException("Resource " + path + " is not a file");
+    }
+    return file;
+  }
+
+  private static String resourceToString(String fileName) throws Exception {
+    ZipFile zipFile = new ZipFile(getResourceFile("/ParseBenchmarkData.zip"));
```

Comment:
Directly reads from the ZIP file; previously it appears user would need to have unzipped it manually.